core/src/test/java/org/apache/calcite/test/RelOptRulesTest.java

```diff
@@ -5582,8 +5603,8 @@ private void checkSemiJoinRuleOnAntiJoin(RelOptRule rule) {
   @Test void testAggregateConstantKeyRule3() {
     final String sql = "select job\n"
         + "from sales.emp\n"
-        + "where sal is null and job = 'Clerk'\n"
-        + "group by sal, job\n"
+        + "where mgr is null and job = 'Clerk'\n"
```

Comment:
I want to know why we need to change the filter condition?

Reply:
@NobiGo Thanks for your reply.
I wanted to submit this pr to solve [ISSUE-5038](https://issues.apache.org/jira/browse/CALCITE-5038) together。
But, I thought about it for a while, I think I may split the changed codes into two prs to prevent ambiguity.
Please review [PR-2742](https://github.com/apache/calcite/pull/2742) firstly.